sql/src/main/java/io/crate/executor/transport/task/UpsertByIdTask.java

```diff
@@ -124,22 +123,17 @@ public void execute(final RowReceiver rowReceiver, Row parameters) {
                 return;
             }
         }
-        Futures.addCallback(result, new FutureCallback<Long>() {
-            @Override
-            public void onSuccess(@Nullable Long result) {
-                rowReceiver.setNextRow(new Row1(result));
+        result.whenComplete(FutureCompleteConsumer.build(
```

Comment:
Generally speaking I don't think the abstraction adds a lot of value.

In this specific case it could be replaced by the OneRowActionListener or with a single function like 

```
        result.whenComplete((count, throwable) -> {
            sendResultToRowReceiver(rowReceiver, count, throwable);
        });
```